### ISSUE TITLE
Cannot git pull: error: Unable to create '/Users/julien/data.neuro/canproco/.git/index.lock': File exists.

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/data-management#321]
```console
julien-macbook:~/data.neuro/canproco $ git annex pull
pull origin 
remote: Énumération des objets: 36448, fait.
remote: Décompte des objets: 100% (36193/36193), fait.
remote: Compression des objets: 100% (23314/23314), fait.
remote: Total 30902 (delta 8180), réutilisés 26587 (delta 4372), réutilisés du pack 0
Receiving objects: 100% (30902/30902), 3.26 MiB | 2.91 MiB/s, done.
Resolving deltas: 100% (8180/8180), completed with 2872 local objects.
From data.neuro.polymtl.ca:datasets/canproco
   5ef3b2120f..c12676d41b  master                            -> origin/master
   23364691d3..0320ba9964  git-annex                         -> origin/git-annex
 * [new branch]            plb/add_sc_seg_for_regionbased    -> origin/plb/add_sc_seg_for_regionbased
 * [new branch]            plb/correct_hc_lesion_mask        -> origin/plb/correct_hc_lesion_mask
 * [new branch]            plb/m12_lesion_seg                -> origin/plb/m12_lesion_seg
 * [new branch]            plb/segment_missing_4_lesion_seg  -> origin/plb/segment_missing_4_lesion_seg
 * [new branch]            plb/update_m12_with_2024_02_26_data -> origin/plb/update_m12_with_2024_02_26_data
   23364691d3..0320ba9964  synced/git-annex                  -> origin/synced/git-annex
   5ef3b2120f..c12676d41b  synced/master                     -> origin/synced/master
 * [new branch]            synced/plb/correct_hc_lesion_mask -> origin/synced/plb/correct_hc_lesion_mask
 * [new branch]            synced/plb/segment_missing_4_lesion_seg -> origin/synced/plb/segment_missing_4_lesion_seg
 * [new branch]            synced/plb/update_m12_with_2024_02_26_data -> origin/synced/plb/update_m12_with_2024_02_26_data
Auto packing the repository in background for optimum performance.
See "git help gc" for manual housekeeping.

Updating 5ef3b2120f..c12676d41b
error: Unable to create '/Users/julien/data.neuro/canproco/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.

Updating 5ef3b2120f..c12676d41b
error: Unable to create '/Users/julien/data.neuro/canproco/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
failed
(merging origin/git-annex into git-annex...)
(recording state in git...)
pull: 1 failed
```

**Comment by mguaypaq** [comment]
Is it possible that some previous `git` or `git annex` command got interrupted in this repository? If there's no currently-running `git` command, you can probably manually delete the `.git/index.lock` file that the error message mentions.

**Comment by jcohenadad** [comment]
Thank you @mguaypaq 

I ran that, and then git pulled, but ran into another error

<details><summary>Terminal output</summary>

```console
julien-macbook:~/data.neuro/canproco $ rm .git/index.lock 
julien-macbook:~/data.neuro/canproco $ git annex pull
pull origin 

Updating 5ef3b2120f..c12676d41b
error: Your local changes to the following files would be overwritten by merge:
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal056/ses-M12/anat/sub-cal056_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal064/ses-M12/anat/sub-cal064_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal072/ses-M12/anat/sub-cal072_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal073/ses-M12/anat/sub-cal073_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal078/ses-M12/anat/sub-cal078_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M12/anat/sub-cal083_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal084/ses-M12/anat/sub-cal084_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal085/ses-M0/anat/sub-cal085_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal085/ses-M0/anat/sub-cal085_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal085/ses-M12/anat/sub-cal085_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal088/ses-M12/anat/sub-cal088_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal090/ses-M12/anat/sub-cal090_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal092/ses-M0/anat/sub-cal092_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal092/ses-M0/anat/sub-cal092_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal092/ses-M12/anat/sub-cal092_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal094/ses-M12/anat/sub-cal094_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal095/ses-M0/anat/sub-cal095_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal0
Aborting

Updating 5ef3b2120f..c12676d41b
error: Your local changes to the following files would be overwritten by merge:
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal056/ses-M12/anat/sub-cal056_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal064/ses-M12/anat/sub-cal064_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal072/ses-M12/anat/sub-cal072_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal073/ses-M12/anat/sub-cal073_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal078/ses-M0/anat/sub-cal078_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal078/ses-M12/anat/sub-cal078_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal080/ses-M0/anat/sub-cal080_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M0/anat/sub-cal083_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal083/ses-M12/anat/sub-cal083_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal084/ses-M0/anat/sub-cal084_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal084/ses-M12/anat/sub-cal084_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal085/ses-M0/anat/sub-cal085_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal085/ses-M0/anat/sub-cal085_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal085/ses-M12/anat/sub-cal085_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal088/ses-M0/anat/sub-cal088_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal088/ses-M12/anat/sub-cal088_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal090/ses-M0/anat/sub-cal090_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal090/ses-M12/anat/sub-cal090_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal092/ses-M0/anat/sub-cal092_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal092/ses-M0/anat/sub-cal092_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal092/ses-M12/anat/sub-cal092_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_STIR_seg-manual.nii.gz
	derivatives/labels/sub-cal094/ses-M0/anat/sub-cal094_ses-M0_T2w_seg-manual.nii.gz
	derivatives/labels/sub-cal094/ses-M12/anat/sub-cal094_ses-M12_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal095/ses-M0/anat/sub-cal095_ses-M0_STIR_labels-disc.nii.gz
	derivatives/labels/sub-cal0
Aborting
failed
pull: 1 failed
```

</details>

This error however seems to be related to local changes that I have not pushed-- since I don't want to push them, I ran git checkout, then pulled, and it seems to have worked:

<details><summary>Terminal output</summary>

```console
julien-macbook:~/data.neuro/canproco $ git checkout .
Updated 4064 paths from the index
julien-macbook:~/data.neuro/canproco $ gb
  git-annex
* master
  synced/master
julien-macbook:~/data.neuro/canproco $ git annex pull
pull origin 

Updating 5ef3b2120f..c12676d41b
Updating files: 100% (3528/3528), done.
Fast-forward
 .bids-validator-config.json                                                                                        |   2 +-
 .bidsignore                                                                                                        |   2 ++
 README.md                                                                                                          |   2 ++
 code/copy_erin_data.py                                                                                             | 103 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 code/split_4D_MT.sh                                                                                                |  61 +++++++++++++++++++++-----------
 code/verify_time_point_difference.py                                                                               | 114 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_labels-disc.nii.gz                                |   2 +-
 derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_STIR_seg-manual.nii.gz                                 |   2 +-
 derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_T2w_seg-manual.json                                    |  35 ++++++++++++++++--
 derivatives/labels/sub-cal056/ses-M0/anat/sub-cal056_ses-M0_T2w_seg-manual.nii.gz                                  |   2 +-
 derivatives/labels/sub-cal056/ses-M12/anat/sub-cal056_ses-M12_STIR_labels-disc.nii.gz                              |   2 +-
 derivatives/labels/sub-cal064/ses-M12/anat/sub-cal064_ses-M12_STIR_labels-disc.nii.gz                              |   2 +-
 derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_labels-disc.nii.gz                                |   2 +-
 derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_STIR_seg-manual.nii.gz                                 |   2 +-
 derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_T2w_seg-manual.json                                    |  35 ++++++++++++++++--
 derivatives/labels/sub-cal072/ses-M0/anat/sub-cal072_ses-M0_T2w_seg-manual.nii.gz                                  |   2 +-
 derivatives/labels/sub-cal072/ses-M12/anat/sub-cal072_ses-M12_STIR_labels-disc.nii.gz                              |   2 +-
 derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_labels-disc.nii.gz                                |   2 +-
 derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_STIR_seg-manual.nii.gz                                 |   2 +-
 derivatives/labels/sub-cal073/ses-M0/anat/sub-cal073_ses-M0_T2w_seg-manual.json                                    |  35 
[...]
ok                                
pull origin 
ok
(recording state in git...)
(recording state in git...)
```

</details>

git log:
```console
c12676d41bd7c2f68980b4db58acb62519204eee (HEAD -> master, origin/synced/master, origin/master, origin/HEAD, synced/master) Merge pull request 'Correction of 4 files' (#4) from plb/correct_4_files into master
0e15677b627862473c438d4ec99378d353169037 added 3 missing json files for manual lesion segmentations
c0a1c914b616b9449e2b201978b8f49445a44bed removed lesion seg of sub-mon052_ses-M0 because too blury
a04d89739c769dc03f23fcda183df62c62f586a9 Merge pull request 'Fix canproco SC segmentations' (#2) from nm/fix_calgary_and_resample into master
261dea6fa925b9b6c6944952855e7b0486074d41 Change dtype to uint8
5e5ac3b17585cde42df549d452673fa2c0ca2bad fix SC mask header and resample
c35b4d889ff2283a825966fd96228790de69aec5 (origin/plb/add_sc_seg_for_regionbased) manually corrected some sc seg added in the previous commit
eba2a01cc13aaed80b5a3b0a26411889a4090bdc added missing sc seg for training of region-based model
558a96d3db7a2e842541100aac783a3f9a146b25 Add script to compare M0 and M12 files
```
